modules/gridmap/editor/grid_map_editor_plugin.cpp

```diff
@@ -951,7 +954,13 @@ void GridMapEditor::update_grid() {
 
 	grid_ofs[edit_axis] = edit_floor[edit_axis] * node->get_cell_size()[edit_axis];
 
-	edit_grid_xform.origin = grid_ofs;
+	// if there's a mesh library and a valid selected tile being used as a cursor, offset the grid, otherwise move it back to the node
+	if (!node->get_mesh_library().is_null() && cursor_instance.is_valid()) {
+		edit_grid_xform.origin = grid_ofs;
+	}
+	else {
+		edit_grid_xform.origin = Vector3();
+	}
```

Comment:
There are some formatting issues:
- Comments should start with a capital letter and end with a period. If the line ends up being too long, it can be worth wrapping manually (I'd suggest wrapping after "used as a cursor," here). The same applies for the two other comments in this PR.
- `else` should be on the same line as the closing bracket of the `if`: `} else {`. You can use clang-format to make sure that your code matches the required style (doesn't apply for comment casing though). See https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html